utils/ProfileProvider.tsx

```diff
@@ -0,0 +1,99 @@
+'use client';
+
+import React, {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { upsertProfile } from '@/api/supabase/queries/profiles';
+import { Profile } from '@/types/schema';
+
+// Define a placeholder user ID for development purposes
+const placeholderUserId = '2abd7296-374a-42d1-bb4f-b813da1615ae';
+
+interface ProfileContextType {
+  profileData: Profile | null;
+  profileReady: boolean;
+  setProfile: (newProfileData: Partial<Profile>) => Promise<void>;
+  loadProfile: () => Promise<void>;
```

Comment:
let's add `has_plot` in `ProfileContextType`, and remove it from the `Profile` type